web3.js/src/transaction.ts

```diff
@@ -133,11 +149,38 @@ export type SignaturePubkeyPair = {
 
 /**
  * List of Transaction object fields that may be initialized at construction
- *
  */
 export type TransactionCtorFields = {
+  /** Optional nonce information used for offline nonce'd transactions */
+  nonceInfo?: NonceInformation | null;
+  /** The transaction fee payer */
+  feePayer?: PublicKey | null;
+  /** One or more signatures */
+  signatures?: Array<SignaturePubkeyPair>;
+  /** A recent blockhash */
+  recentBlockhash?: Blockhash;
+};
+
+/**
+ * List of Transaction object fields that may be initialized at construction
+ */
+export type TransactionBlockhashCtor = {
+  /** The transaction fee payer */
+  feePayer?: PublicKey | null;
+  /** One or more signatures */
+  signatures?: Array<SignaturePubkeyPair>;
   /** A recent blockhash */
-  recentBlockhash?: Blockhash | null;
+  latestBlockhash?: {
+    blockhash: Blockhash;
+    /** the last block chain can advance to before tx is declared expired */
+    lastValidBlockHeight: number;
+  };
+};
+
+/**
+ * List of Transaction object fields that may be initialized at construction
+ */
+export type TransactionNonceCtor = {
   /** Optional nonce information used for offline nonce'd transactions */
   nonceInfo?: NonceInformation | null;
```

Comment:
Same here. It made sense for these to be optional _before_, because `new Transaction(...)` could be called with any combination of these properties. What you're doing by adding function overloads is adding some rigor around what combination of constructor args are and are not allowed.